torchgeo/datasets/bigearthnet.py

```diff
@@ -573,3 +585,441 @@ def plot(
+        self.transforms = transforms
+        self.num_classes = 19
+        self.download = download
+        self.checksum = checksum
+        self.class2idx = {c: i for i, c in enumerate(self.class_set)}
+        self._verify()
+
+        self.metadata_df = pd.read_parquet(os.path.join(self.root, 'metadata.parquet'))
+        self.metadata_df = self.metadata_df[
+            self.metadata_df['split'] == self.split
+        ].reset_index(drop=True)
+
+        # Map chosen classes to ordinal numbers, all others mapped to background class
+        self.ordinal_map = torch.zeros(19)
+        for corine, ordinal in self.clc_codes.items():
+            self.ordinal_map[ordinal] = corine
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.metadata_df)
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        sample: dict[str, Tensor] = {}
+
+        match self.bands:
+            case 's1':
+                sample['image'] = self._load_image(index, 's1')
+            case 's2':
+                sample['image'] = self._load_image(index, 's2')
+            case 'all':
+                sample['image_s1'] = self._load_image(index, 's1')
+                sample['image_s2'] = self._load_image(index, 's2')
+
+        sample['mask'] = self._load_map(index)
+        sample['label'] = self._load_target(index)
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_image(self, index: int, sensor: str) -> Tensor:
+        """Generic image loader for both S1 and S2.
+
+        Args:
+            index: index to return
+            sensor: 's1' or 's2'
+
+        Returns:
+            the sensor image
+        """
+        row = self.metadata_df.loc[index]
+        id_field = 's1_name' if sensor == 's1' else 'patch_id'
+        patch_id = row[id_field]
+        if sensor == 's2':
+            patch_dir = '_'.join(patch_id.split('_')[0:-2])
+        else:
+            patch_dir = '_'.join(patch_id.split('_')[0:-3])
+
+        paths = glob.glob(
+            os.path.join(
+                self.root,
+                self.metadata_locs[sensor]['directory'],
+                patch_dir,
+                patch_id,
+                '*.tif',
+            )
+        )
+
+        if sensor == 's2':
+            paths = sorted(paths, key=sort_sentinel2_bands)
+        else:
+            paths = sorted(paths)
+
+        images = []
+        for path in paths:
+            with rasterio.open(path) as dataset:
+                array = dataset.read(
+                    indexes=1,
+                    out_shape=self.image_size,
+                    out_dtype='int32',
```

Comment:
Should we load as float32 since we're later converting to float32 anyway? Or just load in the original dtype?

Reply:
It looks like the raw data is uint16? int32 doesn't make much sense, no reason to cast dtype multiple times.

Reply:
I don't think pytorch has uint16 datatype, so I am guessing that is why it is being read as int32 so that the tensor can be built and then cast to float32